src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -518,7 +520,7 @@ private function createTypeAlternatives(TypedReference $reference)
         } elseif ($reference->getRequiringClass() && !$reference->canBeAutoregistered() && !$this->strictMode) {
             return ' It cannot be auto-registered because it is from a different root namespace.';
         } else {
-            return;
+            return null;
```

Comment:
The result is used in an `sprintf()` call. Should we rather return an empty string instead?

Reply:
let's do yes